### PR TITLE
477 | Enable ALB access logs for EKS clusters

### LINKED ISSUE #477: Enhancement | Enable ALB access logs for EKS clusters
## Describe the Feature

Implement the required resources to store the logs and enable ALB access logs for apps-prd EKS clusters by default and optionally apps-devstg.

## Expected Behavior

n/a

## Use Case

n/a

## Describe Ideal Solution

n/a

## Alternatives Considered

Probably a good implementation would entail using the [terraform-aws-s3-log-storage](https://github.com/binbashar/terraform-aws-s3-log-storage) module to deploy the storage resources and configure the load balancer via [this annotation](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#custom-attributes) in [this ingress resource](https://github.com/binbashar/le-tf-infra-aws/blob/master/apps-devstg/us-east-1/k8s-eks/k8s-components/networking-ingress.tf#L63-L85).

## Additional Context

n/a

### PR DESCRIPTION
**Pull request by Franr**
## What?
We are adding a variable to create a bucket and enable access logs on the EKS ALB.

## Why?
For security/traceability.

## References
closes #477

### CODE REVIEW

**Comment by exequielrafaela** [comment]
@Franr Cisco there's anything else pending before we can change this PR from draft? Let's us know and we could review it when needed.

**Comment by Franr** [comment]
> @Franr Cisco there's anything else pending before we can change this PR from draft? Let's us know and we could review it when needed.

yes, I was not able to bring up the cluster to be able to test it... there were some errors and I desisted.
If the is another cluster up (like demoapps), I could try there!

**Comment by Franr** [comment]
@exequielrafaela alright Exe, I was finally able to test it! Moving the PR into ready to review.

**Comment by exequielrafaela** [comment]
@Franr small one related to CI test for terraform canonical format => https://app.circleci.com/pipelines/github/binbashar/le-tf-infra-aws/3095/workflows/b22009a1-7f14-47b9-b243-b51deb4be59a/jobs/3876 easily fixed with `leverage terraform format` cmd 💪🏼

File: apps-devstg/us-east-1/k8s-eks-demoapps/k8s-components/networking-ingress.tf [hunk 1]
```diff
@@ -82,6 +82,8 @@ resource "kubernetes_ingress_v1" "apps" {
       # Filter traffic by IP addresses
       # NOTE: this is highly recommended when using an internet-facing ALB
       "alb.ingress.kubernetes.io/inbound-cidrs" = "0.0.0.0/0"
+      # ALB access logs
+      "alb.ingress.kubernetes.io/load-balancer-attributes" = "access_logs.s3.enabled=${var.enable_eks_alb_logging},access_logs.s3.bucket=${var.project}-${var.environment}-alb-logs,access_logs.s3.prefix=eks-cluster-demoapps-ingress"
```
Comment (diego-ojeda-binbash) [inline apps-devstg/us-east-1/k8s-eks-demoapps/k8s-components/networking-ingress.tf:86]: I think this `access_logs.s3.prefix=eks-cluster-demoapps-ingress` should also be defined through a variable.

**Review by diego-ojeda-binbash — CHANGES_REQUESTED** [review]

File: apps-devstg/us-east-1/k8s-eks/identities/ids_external_secrets.tf [hunk 2]
```diff
@@ -51,7 +51,7 @@ resource "aws_iam_policy" "external_secrets_secrets_manager" {
         "kms:DescribeKey"
       ],
       "Resource": [
-        "${data.terraform_remote_state.shared.outputs.aws_kms_key_arn}"
+        "${data.terraform_remote_state.shared-keys.outputs.aws_kms_key_arn}"
```
Comment (Franr) [inline apps-devstg/us-east-1/k8s-eks/identities/ids_external_secrets.tf:54]: a bug I found while creating the identities of the cluster, the state is called `shared-keys`, not `shared`

**Review by diego-ojeda-binbash — APPROVED** [review]

**Review by diego-ojeda-binbash — APPROVED** [review]

**Pull request merged**